### DISCUSSION TITLE
How to sned a private message to another Git

### DISCUSSION THREAD

**Discussion by Exaercase** [community/community#22462] [New to GitHub]
<p>Hello,</p>
<p>   I’m a noob in GitHub and i try to update code of a project outdated but i need help and authorizaton of the owner how could i contact it i don’t find how to send a PM.</p>
<p>Thx in advance for any help</p>
<p>Regards, ExA</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There isn’t a method to send a repository owner a PM via GitHub. You can check to see if they have a public email address listed on their profile or you can open an issue on their repository.</p>
<p>Let us know if you have more questions.</p>

**Comment by Exaercase** [comment]
<blockquote>
<p><a class="mention" href="/lee-dohm">@lee-dohm</a> wrote:</p>
</blockquote>
<blockquote>
<p>There isn’t a method to send a repository owner a PM via GitHub. You can check to see if they have a public email address listed on their profile or you can open an issue on their repository.</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>Let us know if you have more questions.</p>
</blockquote>
<p>Thx,</p>
<p>   It’s All for the momment</p>
<p>Best regard’s</p>
<p>And thx again for answer :slight_smile:</p>
<p>ExA</p>

**Comment by frivoal** [comment]
<p>For technical conversations, staying in the original issue is usually the best, so it is acceptable that there’s no way to DM someone.</p>
<p>However, when someone is misbehaving, doing things that are inappropriate per the Code of Conduct of the Project, it would be very useful to be able to tell them “off the record”. Discussions in public are not often the best ways to solve conflict. Maybe this kind of DM could be restricted to Repo owners or something, but still, something like that would be useful.</p>

**Comment by madscijr** [comment]
<p>There really ought to be a private message feature. I think that would increase the usefulness of github for a lot of people.</p>

**Comment by miallo** [comment]
<p>If you don’t find an email address in their profile, you could clone a repo they have committed to and in <code>git log</code> you can then see their email address…</p>

**Comment by kingthorin** [comment]
<p>Unless they use a noreply address :wink:</p>
<p>You can also just go to any commit in their repo and append <code>.patch</code></p>
<p>Ex:  <a href="https://github.com/OWASP/wstg/commit/60846b32606e2faca28b8b1a4a98896a7c050ff2" rel="noopener nofollow ugc">https://github.com/OWASP/wstg/commit/60846b32606e2faca28b8b1a4a98896a7c050ff2</a>  -&gt; <a href="https://github.com/OWASP/wstg/commit/60846b32606e2faca28b8b1a4a98896a7c050ff2.patch" rel="noopener nofollow ugc">https://github.com/OWASP/wstg/commit/60846b32606e2faca28b8b1a4a98896a7c050ff2.patch</a></p>

**Comment by mathze** [comment]
<p>Especially in context of reporting and processing an abuse against the projects CoC this would be a great feature.<br>
As <a class="mention" href="/frivoal">@frivoal</a> already stated, such kinds of discussions does not belong in public space.<br>
Therefor such a feature would be really useful. Not only for the administrators of a project but also for the users.<br>
What I could imagine is something like a “report suspicion of abuse to admins” flag a user can hit on a comment. Next the user can add an additional short message about why (s)he think this is an abuse. project admins get a notification in their “in-box”. Project admin can check the suspicion and if confirmed, (s)he can contact the “abusing” user with a private message.</p>
<p>Why do I think this feature is also required?<br>
Beside above reason, project admins do not have a “direct” way to talk to an abusing user, if (s)he use a no-reply address, other than public discussion or by contacting the gh-support.<br>
In my opinion, gh-support has more important things to do;). But more important is, that most CoCs (including github suggested one) contain clauses like</p>
<blockquote>
<p>[…]<br>
<em>Consequence</em>: A <strong>private</strong>, written warning from community leaders, providing clarity around the nature of the violation and an explanation of why the behavior was inappropriate. A public apology may be requested.<br>
[…]</p>
</blockquote>
<p>So long story short :slight_smile:, please add this feature</p>

**Comment by YHVHvx** [comment]
<aside class="quote no-group" data-username="frivoal" data-post="4" data-topic="2536" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 frivoal:</div>
<blockquote>
<p>For technical conversations, staying in the original issue is usually the best, so it is acceptable that there’s no way to DM someone.</p>
</blockquote>
</aside>
<p>Sometimes people can archive their project and the issue discussion cant be accessed through Github interface, so i think DM here can be real solution for non active hub, IMHO.</p>

**Comment by thegoatherder** [comment]
<p>Let’s say I’m a fan of someone’s work, it’s in the space I’m working in… I’m using their library, it’s great but I need some of their help with the implementation.</p>
<p>I’d be willing to PAY for their help on a consultation basis. But I have absolutely no way of contacting them to ask them, unless I do it in a public forum e.g. file an issue.  Then my commercial request to them is public, where neither party may have wanted it public and I can’t supply my contact details to them without exposing them publicly either.</p>
<p>Surely this is a major use case to open up DM?</p>

**Comment by jfullstackdev** [comment]
<p>raising an issue and creating a discussion ( GitHub Discussion ) , the new feature, can be equivalent to that PM,</p>
<p>as of exposing details, say of payment, I think, it’s clear that there is the GitHub Sponsors, so all is set, or if there is none, and you still need to expose the details, you can mention him, and agree how to talk privately, maybe via another service,</p>
<p>because I think it’s clear that when someone does not expose his email publicly, he just want privacy, if he wants to be paid, surely, even without the GitHub Sponsors, he will mention it how</p>

**Comment by kingthorin** [comment]
<p>Agreed if people want to be contactable they’ll be contactable.</p>

**Comment by bbkwtx** [comment]
brady dot bass at gmail dot com
